packages/lab-extension/src/index.ts

```diff
@@ -74,14 +81,40 @@ const interfaceSwitcher: JupyterFrontEndPlugin<void> = {
     const { commands, shell } = app;
     const baseUrl = PageConfig.getBaseUrl();
     const trans = translator.load('notebook');
-    const overflowOptions = {
-      overflowMenuOptions: { isVisible: false },
-    };
-    const menubar = new MenuBar(overflowOptions);
+    const nbClassicInstalled =
+      PageConfig.getOption('nbclassic_installed') === 'true';
     const switcher = new Menu({ commands });
-    switcher.title.label = trans.__('Open in...');
-    switcher.title.icon = caretDownIcon;
-    menubar.addMenu(switcher);
+    const switcherOptions: ISwitcherChoice[] = [];
+
+    if (!notebookShell) {
+      switcherOptions.push({
+        command: CommandIDs.openNotebook,
+        commandLabel: trans.__('Notebook'),
+        commandDescription: trans.__('Open in %1', 'Jupyter Notebook'),
+        buttonLabel: 'openNotebook',
+        urlPrefix: `${baseUrl}tree/`,
+      });
+    }
+
+    if (!labShell) {
+      switcherOptions.push({
+        command: CommandIDs.openLab,
+        commandLabel: trans.__('JupyterLab'),
+        commandDescription: trans.__('Open in %1', 'JupyterLab'),
+        buttonLabel: 'openLab',
+        urlPrefix: `${baseUrl}doc/tree/`,
+      });
+    }
+
+    if (nbClassicInstalled) {
+      switcherOptions.push({
+        command: CommandIDs.openNbClassic,
+        commandLabel: trans.__('NbClassic'),
+        commandDescription: trans.__('Open in %1', 'NbClassic'),
+        buttonLabel: 'openNbClassic',
+        urlPrefix: `${baseUrl}nbclassic/tree/`,
```

Comment:
Looks like this still redirects to `notebooks/`?

https://github.com/jupyter/notebook/assets/591645/18487409-3dbb-428e-9afd-477c75f176f7

Not sure if it's an issue in one of the `notebook` or `nbclassic` handlers.

Reply:
Fixed! It was due to incorrect urlprefix! Used `nbclassic/notebooks/` instead of `nbclassic/notebooks`. Thank you!